packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -1304,6 +1315,26 @@ async function startWatcher(opts: SetupOpts) {
             type: HMR_ACTIONS_SENT_TO_BROWSER.TURBOPACK_CONNECTED,
           }
           client.send(JSON.stringify(turbopackConnected))
+
+          const errors = []
+          for (const pageIssues of issues.values()) {
```

Comment:
I feel like we should only send errors for the current page. Maybe we can send them on the first `ping` event instead. There we would have access to the page.

Reply:
The ping event comes too late -- we need to have the issues at the time the error overlay renders.

Reply:
Could we put a header on the WebSocket request that tells the current page and update the issues on ping if the page changes?